instrumentation/tests/dotnet/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+ARG TARGETARCH
+WORKDIR /source
+
+# copy csproj and restore as distinct layers
+COPY *.csproj .
+RUN dotnet restore -a ${TARGETARCH}
```

Comment:
Instead of passing the architecture as a parameter you can use `--use-current-runtime` same for `dotnet publish`

Reply:
I just tried that and it broke, most likely because we run arm on qemu, and the program gets confused and runs with amd64 when it detects the current runtime.

Reply:
Hum I'm thinking if this is actually a bug for `dotnet` CLI 🤔. Anyway, if works as it is...